front/src/common/InfraSelector/InfraSelectorModalBodyStandard.tsx

```diff
@@ -38,13 +45,16 @@ export default function InfraSelectorModalBodyStandard({
 }: InfraSelectorModalBodyStandardProps) {
   const { t } = useTranslation(['translation', 'infraManagement']);
   const dispatch = useDispatch();
-  const infraID = useSelector(getInfraID);
+  const { mode } = useOsrdContext();
+  const updateInfraID = useUpdateInfraID();
+  const infraID = useInfraID();
   const { closeModal } = useContext(ModalContext);
+  const { deleteItinerary } = useOsrdConfActions();
 
   function setInfraID(id: number) {
     dispatch(updateInfraID(id));
     if (onInfraChange) onInfraChange(id);
-    dispatch(deleteItinerary());
+    if (mode === (MODES.simulation || MODES.stdcm)) dispatch(deleteItinerary());
```

Comment:
Are you sure this condition is valid ? I'm not sure it will work if we are on stdcm mode.

Some alternatives:
- `mode === MODES.simulation || mode === MODES.stdcm`
- `[MODES.simulation, MODES.stdcm].includes(mode)`

Reply:
Done